test/classes.coffee

```diff
@@ -738,21 +721,21 @@ test "#2599: other typed constructors should be inherited", ->
   ok (new Derived) not instanceof Base
   ok (new Base) not instanceof Base
 
-test "#2359: extending native objects that use other typed constructors requires defining a constructor", ->
```

Comment:
So, from what I can tell, the main reason ES classes have been implemented the way they are is to fully support extending native objects. The test here therefore became invalid, i.e. you no longer need a constructor to extend native objects.

I decided to adjust the test to check that native objects could be extended with and without a constructor, rather than removing it.